testing/test-runners/baseline_images.py

```diff
@@ -0,0 +1,161 @@
+#!/usr/bin/env python
+
+import argparse
+import md5
+import os
+import requests
+import subprocess
+import time
+
+
+def generate_baselines(args):
+    """
+    Generate baseline image tarball by running ctest in the build directory.
+
+    :param args: a dictionary of arguments, including:
+        xvfb: if True, run ctest within xvfb-run.
+        build: the build directory; created if it does not exist.
+        verbose: the verbosity level.
+    """
+    cwd = os.path.abspath(os.path.expanduser(os.getcwd()))
+    buildPath = os.path.abspath(os.path.expanduser(args.get('build')))
+    if args['verbose'] >= 1:
+        print('Build path: %s' % buildPath)
+    if not os.path.exists(buildPath):
+        os.makedirs(buildPath)
+    if not os.path.isdir(buildPath):
+        raise Exception('build path is not a directory')
+    os.chdir(buildPath)
+    if not os.path.exists(os.path.join(buildPath, 'CMakeFiles')):
+        cmd = ['cmake', '-D', 'FFHEADLESS_TESTS=ON', cwd]
+        if args['verbose'] >= 1:
+            print('Running cmake: %s' % subprocess.list2cmdline(cmd))
+        subprocess.check_call(cmd)
+    tarPath = os.path.join(buildPath, 'base-images.tgz')
+    if os.path.exists(tarPath):
+        os.unlink(tarPath)
+    cmd = ['ctest', '-C', 'baseline_images', '-R', 'baseline_images',
+           '--output-on-failure']
+    if args.get('xvfb'):
+        cmd = ['xvfb-run', '-s', '-ac -screen 0 1280x1024x24', 'bash', '-c',
+               subprocess.list2cmdline(cmd)]
+    if args['verbose'] >= 1:
+        print('Running ctest: %s' % subprocess.list2cmdline(cmd))
+    subprocess.check_call(cmd)
+    os.chdir(cwd)
+    tarSize = os.path.getsize(tarPath)
+    if args['verbose'] >= 1:
+        print('Created baseline image tgz file, %d bytes' % tarSize)
+
+
+def upload_baselines(args):
+    """
+    Upload the baseline image tarball to a girder instance.
+
+    :param args: a dictionary of arguments, including:
+        build: the build directory where the tarball is located.
+        dest: the root url of the Girder instance.
+        folder: the Girder folder ID to upload to.
+        token: a Girder authentication token.
+        verbose: the verbosity level.
+    """
+    buildPath = os.path.abspath(os.path.expanduser(args.get('build')))
+    tarPath = os.path.join(buildPath, 'base-images.tgz')
+    tarSize = os.path.getsize(tarPath)
+    # Get the folder we want to upload to to ensure it exists
+    apiRoot = args['dest'].rstrip('/') + '/api/v1/'
+    headers = {}
+    if args.get('token'):
```

Comment:
I think it might be better just to use girder-client here so we can use an apikey or prompt for user/password.  It should simplify much of this function as well.